### PR TITLE
none:  Skip checkHelmTiller if socat is not installed

### LINKED ISSUE #7197: TestAddons/parallel/HelmTiller: unable to do port forwarding: socat not found
as seen in https://storage.googleapis.com/minikube-builds/logs/7173/67bd258/none_Linux.html#fail_TestAddons%2fparallel%2fHelmTiller
or here
https://storage.googleapis.com/minikube-builds/logs/7202/0478f2d/none_Linux.html#fail_TestAddons%2fparallel%2fHelmTiller

```
addons_test.go:264: tiller-deploy stabilized in 9.093254ms
addons_test.go:266: (dbg) TestAddons/parallel/HelmTiller: waiting 6m0s for pods matching "app=helm" in namespace "kube-system" ...
helpers.go:279: "tiller-deploy-58bf6f4995-2g2zk" [da6a1bec-6cd6-4af8-bbd5-0cadfcfa0f40] Running
addons_test.go:266: (dbg) TestAddons/parallel/HelmTiller: app=helm healthy within 5.007617024s
addons_test.go:273: (dbg) Run:  kubectl --context minikube run --rm helm-test --restart=Never --image=alpine/helm:2.16.3 -it --namespace=kube-system --serviceaccount=tiller -- version
addons_test.go:273: (dbg) Non-zero exit: kubectl --context minikube run --rm helm-test --restart=Never --image=alpine/helm:2.16.3 -it --namespace=kube-system --serviceaccount=tiller -- version: exit status 1 (5m1.69791182s)
-- stdout --
Client: &version.Version{SemVer:"v2.16.3", GitCommit:"1ee0254c86d4ed6887327dabed7aa7da29d7eb0d", GitTreeState:"clean"}
E0324 07:18:29.181700       1 portforward.go:400] an error occurred forwarding 35211 -> 44134: error forwarding port 44134 to pod 8a82d25e107dd63e8593abbf32b5a61060491a199e97a66d57c6bbe7458eaec9, uid : unable to do port forwarding: socat not found
pod "helm-test" deleted
-- /stdout --
** stderr ** 
Unable to use a TTY - input is not a terminal or the right kind of file
If you don't see a command prompt, try pressing enter.
Error attaching, falling back to logs: 
pod kube-system/helm-test terminated (Error)
** /stderr **
```

### PR DESCRIPTION
**Pull request by tstromberg**
Fixes #7197

### CODE REVIEW

File: test/integration/addons_test.go [hunk 1]
```diff
@@ -267,9 +267,17 @@ func validateHelmTillerAddon(ctx context.Context, t *testing.T, profile string)
 		t.Fatalf("wait: %v", err)
 	}
 
+	if NoneDriver() {
+		_, err := exec.LookPath("socat")
```
Comment (medyagh) [inline test/integration/addons_test.go]: How about put a nice warning message for users too ? That they need this ?

Comment (tstromberg) [reply test/integration/addons_test.go]: Honestly, it seems like a pretty invasive peek into implementation details for kubectl. I can't reliably determine which versions of kubectl will need socat on which platforms.

Comment (medyagh) [reply test/integration/addons_test.go]: or how about installing it on our integraiton machines?

**Pull request merged**